### ISSUE TITLE
HE_MeshCollection has null pointer while saving meshes

### ISSUE THREAD

**Issue by nosarious** [wblut/HE_Mesh#70]
I am saving meshes from a mesh collection but sometimes there is a null pointer exception, as if a given mesh cannot be completed. I'm not sure how to test this as try/catch doesn't seem to work, and I cannot test the getMesh(i).

this is the process, based off the example for generating random shapes.

I have a shape which is used to generate voronoi cells. this shape has an inner shape so the voronoi cells are generated with complex caps. The points used to generate these cells come from a dodecahedron, so there are usually more than thirty.

For some of these shapes I am creating wireframes from the cells returned as a mesh collection, creating another mesh collection of wireframe cells. 

during the creation process and drawing process I am getting n o errors about cells not existing, but when I step through the mesh collection to save I will sometimes get a null pointer exception. 

 ```
for (int i=0; i<allFirstWireFrames.size(); i++){
    HET_Export.saveToOBJ(allFirstWireFrames.getMesh(i), firstFolder,"firstWireframeCell"+i);//null pointer
 }
```

I cannot test if a mesh exists in a mesh collection without getting the mesh which causes a null pointer exception since it  doesn't exist...

**Comment by nosarious** [comment]
while running this again, I got the following error, so perhaps it is throwing a null pointer somewhere but I'm unsure how to catch and avoid it. Hopefully this helps?

asteroid has 70 cells.first shape has 70 cells.dual has 70 cells.
java.lang.NullPointerException
	at wblut.hemesh.HES_CatmullClark.apply(HES_CatmullClark.java:124)
	at wblut.hemesh.HE_Mesh.subdivide(HE_Mesh.java:170)
	at ObjectSlicing_15_save.createWireframe(ObjectSlicing_15_save.java:550)
	at ObjectSlicing_15_save.prepSettings(ObjectSlicing_15_save.java:157)
	at ObjectSlicing_15_save.mousePressed(ObjectSlicing_15_save.java:255)
	at processing.core.PApplet.mousePressed(PApplet.java:2757)
	at processing.core.PApplet.handleMouseEvent(PApplet.java:2690)
	at processing.core.PApplet.dequeueEvents(PApplet.java:2607)
	at processing.core.PApplet.handleDraw(PApplet.java:2448)
	at processing.opengl.PSurfaceJOGL$DrawListener.display(PSurfaceJOGL.java:849)
	at jogamp.opengl.GLDrawableHelper.displayImpl(GLDrawableHelper.java:692)
	at jogamp.opengl.GLDrawableHelper.display(GLDrawableHelper.java:674)
	at jogamp.opengl.GLAutoDrawableBase$2.run(GLAutoDrawableBase.java:443)
	at jogamp.opengl.GLDrawableHelper.invokeGLImpl(GLDrawableHelper.java:1293)
	at jogamp.opengl.GLDrawableHelper.invokeGL(GLDrawableHelper.java:1147)
	at com.jogamp.newt.opengl.GLWindow.display(GLWindow.java:759)
	at com.jogamp.opengl.util.AWTAnimatorImpl.display(AWTAnimatorImpl.java:81)
	at com.jogamp.opengl.util.AnimatorBase.display(AnimatorBase.java:452)
	at com.jogamp.opengl.util.FPSAnimator$MainTask.run(FPSAnimator.java:178)
	at java.util.TimerThread.mainLoop(Timer.java:555)
	at java.util.TimerThread.run(Timer.java:505)
java.lang.NullPointerException
	at wblut.hemesh.HES_CatmullClark.apply(HES_CatmullClark.java:124)
	at wblut.hemesh.HE_Mesh.subdivide(HE_Mesh.java:170)
	at ObjectSlicing_15_save.createWireframe(ObjectSlicing_15_save.java:550)
	at ObjectSlicing_15_save.prepSettings(ObjectSlicing_15_save.java:157)
	at ObjectSlicing_15_save.mousePressed(ObjectSlicing_15_save.java:255)
	at processing.core.PApplet.mousePressed(PApplet.java:2757)
	at processing.core.PApplet.handleMouseEvent(PApplet.java:2690)
	at processing.core.PApplet.dequeueEvents(PApplet.java:2607)
	at processing.core.PApplet.handleDraw(PApplet.java:2448)
	at processing.opengl.PSurfaceJOGL$DrawListener.display(PSurfaceJOGL.java:849)
	at jogamp.opengl.GLDrawableHelper.displayImpl(GLDrawableHelper.java:692)
	at jogamp.opengl.GLDrawableHelper.display(GLDrawableHelper.java:674)
	at jogamp.opengl.GLAutoDrawableBase$2.run(GLAutoDrawableBase.java:443)
	at jogamp.opengl.GLDrawableHelper.invokeGLImpl(GLDrawableHelper.java:1293)
	at jogamp.opengl.GLDrawableHelper.invokeGL(GLDrawableHelper.java:1147)
	at com.jogamp.newt.opengl.GLWindow.display(GLWindow.java:759)
	at com.jogamp.opengl.util.AWTAnimatorImpl.display(AWTAnimatorImpl.java:81)
	at com.jogamp.opengl.util.AnimatorBase.display(AnimatorBase.java:452)
	at com.jogamp.opengl.util.FPSAnimator$MainTask.run(FPSAnimator.java:178)
	at java.util.TimerThread.mainLoop(Timer.java:555)
	at java.util.TimerThread.run(Timer.java:505)

**Comment by nosarious** [comment]
gad, I feel like an idiot.

I had been using a try/catch for a catmull subdivision, which created the error code above. Removing that enabled the try/catch during the save routine to work properly.

```
for (int i=0; i<allFirstWireFrames.size(); i++){
      try {
        HET_Export.saveToOBJ(allFirstWireFrames.getMesh(i), firstFolder,"firstWireframeCell"+i);
      }
      catch(final Exception ex) {
        //oops HE_Mesh messed up, retreat!
        ex.printStackTrace();
        print (" failed to create a file for ");//thisNewMesh=tmp;
      }
      print(" "+i);
  }
```

sometimes my code offends even me.

**Comment by wblut** [comment]
The save routine failed to properly check fo null objects. This issue will be fixed in the next release 5.1.1
